### PR TITLE
I2C - add functionality to change i2c clock speed

### PR DESCRIPTION
**Pull request by bigdinotech**
-adds functionality to change clock speed
-clock speed is set/changed when Wire.begin() is called
-default is standard mode
-call Wire.begin(I2C_SPEED_SLOW) for standard speed (100k)
-call Wire.begin(I2C_SPEED_FAST) for full speed (400k)

### CODE REVIEW

**Comment by bigdinotech** [comment]
@SidLeung @kmsywula ready for code review

**Comment by bigdinotech** [comment]
linked to #170

File: libraries/Wire/src/Wire.h [hunk 1]
```diff
@@ -24,12 +24,15 @@
...
-#define BUFFER_LENGTH 32
+#define BUFFER_LENGTH   32
+#define I2C_SPEED_SLOW  1
+#define I2C_SPEED_FAST  2
 
 class TwoWire : public Stream {
 public:
 	TwoWire(void);
 	void begin();
+    void begin(int);
```
Comment (eriknyquist) [inline libraries/Wire/src/Wire.h]: indentation mismatch- perhaps 4 spaces were used instead of a tab?

**Comment by eriknyquist** [comment]
@bigdinotech is this blocked by #166?

**Comment by bigdinotech** [comment]
@eriknyquist yes. And there is a couple more fixes that will be on top of this

**Comment by bigdinotech** [comment]
This can be closed.
I created a new PR https://github.com/01org/corelibs-arduino101/pull/224

**Pull request closed without merging**